lib/src/code/code.dart

```diff
@@ -57,11 +58,23 @@ class Code {
     final issues = <Issue>[];
     final List<FoldableBlock> foldableBlocks;
 
+    final lines = CodeLineBuilder.textToCodeLines(
+      text: text,
+      highlighted: highlighted,
+      commentsByLines: commentParser.getCommentsByLines(),
+    );
+
     if (highlighted == null) {
       foldableBlocks = const [];
     } else {
-      final parser = HighlightFoldableBlockParser();
-      parser.parse(highlighted, serviceCommentsNodesSet);
+      final parser = FoldableBlockParserFactory.provideParser(language ?? java);
```

Comment:
Unfortunately If we don't know about language at this place, I think it will be good idea to provide parser, which covers most of language we supports in playground (Java, Dart, Scala. May be Go, not sure). Of course this package can be used in another app, so exact we can do is make `mode` parameter at `Code` non nullable& I think it will be more clear solution

Reply:
Let's not do any parsing and folding at all without the language info. No folding is better than a wrong folding.